client/packages/system/src/Patient/ProgramEnrolment/Components/ListView.tsx

```diff
@@ -27,11 +27,9 @@ const programAdditionalInfoAccessor: ColumnDataAccessor<
   ProgramEnrolmentRowFragmentWithId,
   string[]
 > = ({ rowData }): string[] => {
-  const additionalInfo = [];
-
-  if (rowData?.activeProgramEvents[0]?.data) {
-    additionalInfo.push(rowData.activeProgramEvents[0].data);
-  }
+  const additionalInfo = rowData.activeProgramEvents
+    .map(e => (e.type?.startsWith('status') ? e.data : undefined))
+    .filter((data): data is string => !!data);
```

Comment:
For me, the logic is clearer if you are explicit with the filter and then map the data to another shape. It also means that the mapping is working on the restricted data set.. which yeah, is only a handful of elements, so makes no difference.

```suggestion
  const additionalInfo = rowData.activeProgramEvents
    .filter(event => event.type?.startsWith('status') && !!event.data)
    .map(event => String(event.data));
```

Reply:
Thanks, did this now since I didn't like the runtime String conversion...

```
export const getStatusEventData = (events: ProgramEventFragment[]): string[] =>
  events
    .filter(e => e.type?.startsWith('status'))
    .map(e => e.data)
    .filter((data): data is string => !!data);
```

Reply:
sure - I thought that given the `data` property has a type of `string`, then the compile time check wasn't adding much. there is a possibility that someone will change the type though and the compiler won't alert you.
Perhaps the original version of `map` then `filter` is neater then? I can't see a way to have the compile time check without the final filter line

Reply:
Think its the most readable version now (?) and since I doubt it will be a performance issue I leave it :)